contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol

```diff
@@ -0,0 +1,708 @@
+// ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠛⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡄⠀⠀⠀⠀⠀⠀⠀⠀⠀⢰⣿⣿⣿⣿⠃⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠘⣿⣿⣿⣿⣧⠀⠀⢸⣿⣿⣿⣗⠀⠀⠀⢸⣿⣿⣿⡯⠀⠀⠀⠀⠹⢿⣿⣿⣿⣿⣾⣾⣷⣿⣿⣿⣿⡿⠋⠀⠀⠀⠀
+// ⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠙⠙⠋⠛⠙⠋⠛⠙⠋⠛⠙⠋⠃⠀⠀⠀⠀⠀⠀⠀⠀⠠⠿⠻⠟⠿⠃⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠸⠟⠿⠟⠿⠆⠀⠸⠿⠿⠟⠯⠀⠀⠀⠸⠿⠿⠿⠏⠀⠀⠀⠀⠀⠈⠉⠻⠻⡿⣿⢿⡿⡿⠿⠛⠁⠀⠀⠀⠀⠀⠀
+//                    allo.gitcoin.co
+
+/// @title LTIP Simple Strategy
+/// @author @thelostone-mc <aditya@gitcoin.co>, @0xKurt <kurt@gitcoin.co>, @codenamejason <jason@gitcoin.co>, @0xZakk <zakk@gitcoin.co>, @nfrgosselin <nate@gitcoin.co>, @bitbeckers
+/// @notice Strategy for Long-Term Incentive Programs (LTIP) allocation with distribution vested over time. The simple strategy retains the funds in the round untill the period has expired
+contract LTIPSimpleStrategy is BaseStrategy, ReentrancyGuard {
+    /// ================================
+    /// ========== Struct ==============
+    /// ================================
+
+    /// @notice Stores the details of the recipients.
+    struct Recipient {
+        bool useRegistryAnchor;
+        address recipientAddress;
+        uint256 allocationAmount;
+        Status recipientStatus;
+        Metadata metadata;
+    }
+
+    /// @notice Pointer for the vesting plan.
+    struct VestingPlan {
+        address vestingContract;
+        uint256 tokenId;
+    }
+
+    /// @notice The parameters used to initialize the strategy
+    struct InitializeParams {
+        // slot 0
+        bool registryGating;
+        bool metadataRequired;
+        // slot 1
+        uint256 votingThreshold;
+        // slot 2
+        uint64 registrationStartTime;
+        uint64 registrationEndTime;
+        uint64 reviewStartTime;
+        uint64 reviewEndTime;
+        // slot 3
+        uint64 allocationStartTime;
+        uint64 allocationEndTime;
+        uint64 distributionStartTime;
+        uint64 distributionEndTime;
+        // slot 4
+        uint64 vestingPeriod;
+    }
+
+    /// ===============================
+    /// ========== Errors =============
+    /// ===============================
+
+    /// @notice Thrown when the application is cancelled. Cancelled applicants cannot reapply.
+    error APPLICATION_CANCELLED();
+
+    /// @notice Thrown when the recipient already has allocation.
+    error ALREADY_ALLOCATED();
+
+    /// @notice Thrown when the review period has ended.
+    error REVIEW_NOT_ACTIVE();
+
+    /// @notice Thrown when the recipient doesn't have enough votes for fund distribution.
+    error INSUFFICIENT_VOTES();
+
+    /// ===============================
+    /// ========== Events =============
+    /// ===============================
+
+    /// @notice Emitted when a recipient updates their registration
+    /// @param recipientId Id of the recipient
+    /// @param data The encoded data - (address recipientId, address recipientAddress, Metadata metadata)
+    /// @param sender The sender of the transaction
+    event UpdatedRegistration(address indexed recipientId, bytes data, address sender);
```

Comment:
Do you think it's worth indexing the `sender` aswell in the events?

Reply:
Depends, is it usefull for you indexing? We can always get the source of the tx from the event emission, if that is enough I can remove this for sure

Reply:
I am not really sure if this is needed for the indexer, @thelostone-mc might know more

Reply:
Nah i think we can do without it! The indexer can query it from tx source